### PR TITLE
Add port to MQTT json messages. Implements #127.

### PR DESCRIPTION
**Pull request by kersing**
Patch allows setting port on downlink messages.

### CODE REVIEW

File: core/components/handler/pktStorage.go [hunk 1]
```diff
@@ -145,6 +147,7 @@ func (e pktEntry) MarshalBinary() ([]byte, error) {
 	rw := readwriter.New(nil)
 	rw.Write(e.AppEUI)
 	rw.Write(e.DevEUI)
+	rw.Write(uint16(e.FPort))
```
Comment (htdvisser) [inline core/components/handler/pktStorage.go]: This line makes it difficult to deploy this feature to our _staging_ environment, as existing databases can still contain data in the old format. This means we have three options:

- Don't deploy this to staging
- Delete all old data and deploy to staging
- Write something that migrates old data

I vote for option 2, as it is the simplest solution. There aren't that many scheduled packets that would get lost

Comment (kersing) [inline core/components/handler/pktStorage.go]: I vote for option 2 as well as we're currently not able to send any downlink data anyway... So deploying it now would not make a difference.

Comment (htdvisser) [inline core/components/handler/pktStorage.go]: What do you mean with "currently not able to send any downlink data anyway"?
The current implementation just sets the downlink port to 1, which is working just fine.

Comment (kersing) [inline core/components/handler/pktStorage.go]: see http://forum.thethingsnetwork.org/t/downlink-problem/2433 - Seems to be an issue with downlink at the moment.

**Comment by htdvisser** [comment]
We will not merge this PR at this time because of the issues this will cause with the staging environment. Support for a downlink FPort has been implemented in the `refactor` branch (#199) and will be deployed in the production environment.

Nevertheless I would like to thank @kersing for his work.

**Pull request closed without merging**